### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by wiseodd**
1. Updated `curvlinops` to `>= 2.0`. @f-dangel has just made a new release.
2. Move out `asdl` dependency out of the build system. It needs to be installed manually until @kazukiosawa uploaded a new release to PyPi.

@runame please create a new release & upload to PyPi if things here are OK.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ The [code](https://github.com/runame/laplace-redux) to reproduce the experiments
 To install laplace with `pip`, run the following:
 
 ```bash
+pip install git+https://github.com/kazukiosawa/asdl
```
Comment (runame) [inline README.md:59]: Kazuki just created a new PyPI release (0.1a4), so let's use that.

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -21,6 +21,7 @@ jobs:
         run: |
           python -m pip install --upgrade pip 
           pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cpu
+          pip install git+https://github.com/kazukiosawa/asdl
```
Comment (runame) [inline .github/workflows/pytest.yml:24]: can be removed

File: setup.cfg [hunk 3]
```diff
@@ -38,10 +38,9 @@ install_requires =
     torchaudio
     backpack-for-pytorch
     asdfghjkl
-    asdl @ git+https://github.com/kazukiosawa/asdl
```
Comment (runame) [inline setup.cfg:41]: Add `asdfghjkl == 0.1a4`.

**Review by runame — CHANGES_REQUESTED** [review]

**Comment by wiseodd** [comment]
@runame please check

**Comment by aleximmer** [comment]
Alternative that integrates old version of asdl into third_party at #224. Please don't merge either of them until we made a decision on how to best proceed.

**Comment by runame** [comment]
@wiseodd @aleximmer Can this PR be closed?

**Pull request closed without merging**